example/lib/adapter/ble_adapter.dart

```diff
@@ -37,29 +41,35 @@ class BleAdapter {
   BleAdapter._internal(this._bleManager, this._blemulator) {
     _setupSimulation();
     _bleManager.createClient();
+    _setupBlePeripheralsController();
   }
 
-  Stream<BlePeripheral> startPeripheralScan() {
-    return _bleManager.startPeripheralScan().transform(
-      new StreamTransformer<ScanResult, BlePeripheral>.fromHandlers(
-        handleData: (ScanResult scanResult, EventSink<BlePeripheral> sink) {
-          if (scanResult.advertisementData.localName != null) {
-            final peripheral = BlePeripheral(
-              scanResult.peripheral.name,
-              scanResult.peripheral.identifier,
-              scanResult.rssi,
-              false,
-              BlePeripheralCategoryResolver.categoryForName(
-                  scanResult.peripheral.name),
-            );
-            sink.add(peripheral);
-          }
-        },
-      ),
+  void _setupBlePeripheralsController() {
+    _blePeripheralsController = StreamController.broadcast(
```

Comment:
Why a broadcast? Isn't there at most one active subscription at all times?

Reply:
There's only one BleAdapter object in an entire app. If `_blePeripheralsController ` is only called created once in the init of BleAdapter, it has to be a broadcast StreamController, because otherwise, it would not be possible to listen to it again, once scanning was stopped and started again.